src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -308,13 +308,7 @@
            (i18n/trs "Missing:")
            (str/join "\n" missing-files))))
 
-; TODO - PE-5529 - these should be moved to jvm-c-a.
-(schema/defn get-subject :- schema/Str
-  [cert :- Certificate]
-  (-> cert
-      (.getSubjectX500Principal)
-      (.getName)))
```

Comment:
The problem with using `X500Principal.getName()` for this is that the format in which the method returns the subject does not preserve the original ordering of the RDNs as encoded into DER/ASN.1 in the certificate.  The `X500Principal` object does, however, retain a copy of the original encoded bytes from the certificate, available via a call to `X500Principal.getEncoded()`.  BouncyCastle has a helper for converting the DER/ASN.1 encoded bytes back into a string.  It didn't seem appropriate to call into the BouncyCastle helper directly from Puppet Server code.  Instead, then, I wrote a helper in jvm-ssl-utils which uses `X500Principal.getEncoded` and BouncyCastle in combination to return a subject from the `X509Certificate` which has the order of the RDNs preserved from how they were encoded into the source PEM file for the `X509Certificate`.  Throughout the rest of this PR, I changed Puppet Server code to call into the new jvm-ssl-utils helper.